Sponsor us cta

SondraE opened the pull request:
## Related Issue(s)
[#836](https://github.com/oddbird/oddleventy/issues/836)


## Steps to test/reproduce
- [x] Create new Sponsor Us CTA
- [x] Change Hire/Sponsor Us CTA background 
- [x] On Tools add Sponsor Us CTA
- [x] On Work (list & detail) add Hire Us CTA
- [x] On Services (list & detail) add Hire Us CTA
- [x] Remove current CTA tools (list & detail) and replace with Sponsor Us CTA
- [x] Change the phone image in the Hire Us CTA
- [x] Invert Hire/Sponsor Us CTA button background color 
- [x] Hide main/old site CTA on work (list & detail)
- [x] Hide main/old site CTA on service (list & detail)
- [x] Replace work and service footer with one that includes social links
- [x] Add a new demo image to the Sponsor Us CTA
- [x] Link to Open Collective instead of Contact page?

stacyk commented on content/_includes/site/sponsor-us.njk:
```diff
@@ -0,0 +1,23 @@
+      <span class="title-description">That you depend on</span>
+    </h2>
+
+    <a href='/contact/' data-btn="link-button" role="button" class="section-link" draggable="false">Let's chat</a>
```
@mirisuzanne we added the `data-ccs-colors="invert"` to the parent container (aside) but this also inverts the button, which we don't want. If we add `data-ccs-colors` without any value to this button, does that work? It seemed to in my local testing but I wanted to check with you cc: @SondraE

mirisuzanne replied:
From the designs, it looks to me like this section is always light-on-dark? I would expect `data-ccs-colors="dark"` to be the best choice here. And ideally the link-button styles should adapt based on the theme. "Invert" will sometimes be dark-on-light, because it uses the reverse of whatever the user has set for the page.

Aside: why a link with role button, and draggable set to false?

stacyk replied:
> From the designs, it looks to me like this section is always light-on-dark?

@mirisuzanne I think she changed her mind yesterday and wanted to use the site's heading gradient for the background of this callout container, but she wanted to invert it so the background doesn't stand out as much. This also means it should change based on light or dark mode. Maybe we just need to assign new colors for this background instead of trying to use the same foreground color in the opposite color mode. 

No idea on the button question. I think this is copied as-is from the hire-us.njk file

dvdherron replied:
> Aside: why a link with role button, and draggable set to false?

@mirisuzanne That was my doing. Since it's a link styled as a button, I followed [this](https://www.tempertemper.net/blog/when-design-breaks-semantics) and tried to make it behave as much as a button as possible.

mirisuzanne replied:
Hmmm, I'm skeptical for the reasons they mention at the end. If there weren't alternatives, I'd understand. But it seems to me like a better approach would be to create a "high-priority link" style that is still clearly a link, but maybe on a background, instead of looking like a button at all.

stacyk commented on src/scss/layout/_cta.scss:
```diff
@@ -18,7 +18,7 @@
   --stacked-margin-block: var(--shim);
   --title-action-margin-block-end: var(--quarter-shim);
 
-  background-image: url('/assets/media/callout-bg-1.webp');
```
are we using this background image for anything still? If not, we can remove it.

SondraE commented:
@dvdherron 
> Just to be clear, this is mainly waiting on a new (designed?) image in the CTA?

It's these items

- Invert Hire/Sponsor Us CTA button background color (see convo about CTA background change from image to gradient colors)
- Replace work and service footer with one that includes social links (like footer on Homepage)
- Create/Add a new demo image to the Sponsor Us CTA (may or may not be inside the mobile phone)

dvdherron commented:
>     * Invert Hire/Sponsor Us CTA button background color (see convo about CTA background change from image to gradient colors)

@SondraE I'm not sure what the ideal approach is if we want the background gradient to change with the color mode. I think we would have to manually select colors for the button to get what we want to happen to work. Or we could change the CTAs from being inverted to being set in either "dark" or "light" mode until we have a better solution?

> 
>     * Replace work and service footer with one that includes social links (like footer on Homepage)

I fixed this in https://github.com/oddbird/oddleventy/pull/856/commits/ffb931d40a79a7a8d1b054e3470465192416025e

We're still planning to remove the main site CTA in the PR, right?

>     * Create/Add a new demo image to the Sponsor Us CTA (may or may not be inside the mobile phone)

I haven't been able to work on a demo/design that works here. If others have ideas I can give it a shot in the morning. 

Other questions:
- Are we planning to change the text from "Hire Us" to "Sponsor Us" in the sponsor us cta?

- And where exactly should the button-link lead to if we're no longer taking people to the contact page? Just our OC homepage?

SondraE commented:
@dvdherron 

> We're still planning to remove the main site CTA in the PR, right?

Yep. Stacy and I tried to remove the main/old "Want to chat about..." CTA section from Work and Services, but then ran out of time to figure out where that is being inserted into those pages. If you are able to figure that out and remove it, that would be great! (We do want to show the new Hire Us CTA section on the Work and Services pages. I've set those to true.)

> Are we planning to change the text from "Hire Us" to "Sponsor Us" in the sponsor us cta?

Updated

> And where exactly should the button-link lead to if we're no longer taking people to the contact page? Just our OC homepage?

Updated. It now leads to OC homepage

dvdherron commented:
@SondraE I adjusted the layout of screenshot a bit to make it more narrow. If we want the polyfill demo to bleed over the edge of the background in this CTA, I think we will have to make the gradient into a background image with similar dimensions of the background image that it's the Hire Us CTA.

Let me know what you think. 

![image](https://github.com/user-attachments/assets/0105f324-dff7-4b9b-a24e-2719d37ac59e)


![image](https://github.com/user-attachments/assets/d364bdcb-62dd-4b01-94ab-63619ea1cc59)

stacyk commented:
> @SondraE I adjusted the layout of screenshot a bit to make it more narrow. ~If we want the polyfill demo to bleed over the edge of the background in this CTA, I think we will have to make the gradient into a background image with similar dimensions of the background image that it's the Hire Us CTA.~
> 
> Let me know what you think.
> 
> ![image](https://private-user-images.githubusercontent.com/41588129/430114739-1e66e576-2abd-4a25-9dba-559377abdeb5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwNDI0OTEsIm5iZiI6MTc0NDA0MjE5MSwicGF0aCI6Ii80MTU4ODEyOS80MzAxMTQ3MzktMWU2NmU1NzYtMmFiZC00YTI1LTlkYmEtNTU5Mzc3YWJkZWI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDE2MDk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhYTNiNWU2YmY5ZmYyNmM4YjE4NjQyNzk3YjFhNGYyM2YwNzNmNjkyNzc0ODMxOWU5NjhkMDhlYTI1OGJkYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3mojUQA0QzdQOw8PDPi9ggN23lRg0-FSu8RDo1tZJGA)
> 

@dvdherron do you have a link to where this image/code demo is located? I think it would be nice to increase the code text size and maybe crop out some of the bottom space/vertical space in general, which would be a nice enhancement for the small screen audience

dvdherron commented:
> > @SondraE I adjusted the layout of screenshot a bit to make it more narrow. ~If we want the polyfill demo to bleed over the edge of the background in this CTA, I think we will have to make the gradient into a background image with similar dimensions of the background image that it's the Hire Us CTA.~
> > Let me know what you think.
> > ![image](https://private-user-images.githubusercontent.com/41588129/430114739-1e66e576-2abd-4a25-9dba-559377abdeb5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwNDI0OTEsIm5iZiI6MTc0NDA0MjE5MSwicGF0aCI6Ii80MTU4ODEyOS80MzAxMTQ3MzktMWU2NmU1NzYtMmFiZC00YTI1LTlkYmEtNTU5Mzc3YWJkZWI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDE2MDk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhYTNiNWU2YmY5ZmYyNmM4YjE4NjQyNzk3YjFhNGYyM2YwNzNmNjkyNzc0ODMxOWU5NjhkMDhlYTI1OGJkYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3mojUQA0QzdQOw8PDPi9ggN23lRg0-FSu8RDo1tZJGA)
> 
> @dvdherron do you have a link to where this image/code demo is located? I think it would be nice to increase the code text size and maybe crop out some of the bottom space/vertical space in general, which would be a nice enhancement for the small screen audience

yes! the codepen is [here](https://codepen.io/dvdherron/pen/ByavGaE/ebcebcb16370f6976ee6e91c36f04754). I'll bump up the code text size and adjust the vertical spacing to make this fit better.

dvdherron commented:
> @dvdherron do you have a link to where this image/code demo is located? I think it would be nice to increase the code text size and maybe crop out some of the bottom space/vertical space in general, which would be a nice enhancement for the small screen audience

I bumped up the code text size and adjusted some vertical spacing in the demo in my recent commits. I think it's easier to see/read now? Let me know what you think @stacyk 
![image](https://github.com/user-attachments/assets/002b6a15-a0da-44f1-b58d-f2e8c49935b5)

stacyk commented:
> > @dvdherron do you have a link to where this image/code demo is located? I think it would be nice to increase the code text size and maybe crop out some of the bottom space/vertical space in general, which would be a nice enhancement for the small screen audience
> 
> I bumped up the code text size and adjusted some vertical spacing in the demo in my recent commits. I think it's easier to see/read now? 

I agree!

stacyk commented on content/_includes/site/hire-us.njk:
```diff
@@ -1,6 +1,6 @@
 {% import 'embed.macros.njk' as embed %}
 
-<aside data-layout="cta" aria-labelledby="hire-cta-title" class="hire-us-cta" data-ccs-colors="invert">
+<aside data-layout="cta" aria-labelledby="hire-cta-title" class="hire-us-cta" data-ccs-colors="light">
```
@SondraE I noticed that we are currently using the dark gradient on both the light and dark modes. I was out for a week but wanted to check in with you to see if that was a decision you all made while I was out as I believe when we last spoke you wanted the light gradient background on light mode and the dark gradient on dark mode.

SondraE replied:
@stacyk I didn't intend for background to be the same in light and dark mode. The background colors should change the the mode just like the gradient text changes with the mode.

dvdherron replied:
@SondraE @stacyk 

Ok, I think this is adjusted as preferred.
- The background changes based on mode. 
- The colors are the opposite of whatever mode one is in.

Is that what we want?

Light mode:
![image](https://github.com/user-attachments/assets/fe67e18b-c992-492f-b194-a93ceb7bef7d)

Dark mode:
![image](https://github.com/user-attachments/assets/00fd3884-669d-4f0c-b22e-f433b675f927)

And if we think the link underline needs to stand out more we could swap the positions of the `--brand-gradient-end` and `--brand-gradient-start` colors

![image](https://github.com/user-attachments/assets/25517f0d-316f-4b45-a4cd-81e48e951088)

![image](https://github.com/user-attachments/assets/dbbc3f5b-dbab-42df-918f-501121c2294b)

dvdherron commented:
@SondraE @stacyk @mirisuzanne This is ready for another review.

mirisuzanne reviewed (approved):
This looks great. Is it intentional that the oddnews stuff is in this PR?

dvdherron commented:
@SondraE I updated the link underlines to match the links in the footer in https://github.com/oddbird/oddleventy/pull/856/commits/e555f88393132efdf9f287f89a9782e51aa1dd7d Now that you can see it on the page what do you think?

And I removed the oddnews related changes in https://github.com/oddbird/oddleventy/pull/856/commits/6f70713b6fec03c0ea47aa0f1ab38e3f7867ed07 Let me know if I missed anything else. 

![image](https://github.com/user-attachments/assets/4e22fc7e-568c-46fc-8afd-12baf5019ff1)

![image](https://github.com/user-attachments/assets/0a991402-a4d9-44be-95a3-e8a48865c137)

dvdherron commented on content/_data/taxonomy.yaml:
```diff
@@ -21,6 +21,9 @@ post:
     plural: Episodes
+  - tag: OddNews
+    icon: winging-it-lines
+    plural: Issues
```
removed in https://github.com/oddbird/oddleventy/commit/6f70713b6fec03c0ea47aa0f1ab38e3f7867ed07

dvdherron commented on content/tools/tools.11tydata.yaml:
```diff
@@ -1,4 +1,4 @@
 oss: creator
 sponsors: true
-hire_cta: true
+sponsor_cta: true
```
Something I look into. I made a new issue to work on this https://github.com/oddbird/oddleventy/issues/871
